From 184dfc4c0e389a5cc63dae6b8bca9483a10b098b Mon Sep 17 00:00:00 2001
From: "Qi, Yadong" <yadong.qi@intel.com>
Date: Tue, 27 Feb 2024 16:19:19 +0800
Subject: [PATCH] zlib changes for angle build

changes so external angle can reuse this module.

Signed-off-by: Qi, Yadong <yadong.qi@intel.com>
Tracked-On: OAM-115781
---
 Android.bp | 24 ++++++++++++++++++++++--
 1 file changed, 22 insertions(+), 2 deletions(-)

diff --git a/Android.bp b/Android.bp
index 6895ea3..2a6ce7c 100644
--- a/Android.bp
+++ b/Android.bp
@@ -273,19 +273,39 @@ cc_binary {
     },
 }
 
+cc_library {
+    name: "zlib_google_compression_utils_portable",
+    defaults: ["libz_defaults"],
+    srcs: [
+        "google/compression_utils_portable.cc",
+    ],
+    export_include_dirs: ["google"],
+    host_supported: true,
+    shared_libs: ["libz"],
+    sdk_version: "minimum",
+    visibility: ["//external/angle"],
+    apex_available: [
+        "com.android.runtime",
+        "//apex_available:platform",
+    ],
+}
+
 cc_test {
     name: "zlib_tests",
     srcs: [
         "contrib/tests/infcover.cc",
         "contrib/tests/utils_unittest.cc",
-        "google/compression_utils_portable.cc",
+    ],
+    cflags: [
+        "-DCMAKE_STANDALONE_UNITTESTS",
+        "-Wno-unused-parameter",
     ],
     include_dirs: [
-        "external/zlib/google",
         // These tests include "gtest.h" rather than the usual "gtest/gtest.h".
         "external/googletest/googletest/include/gtest/",
     ],
     shared_libs: ["libz"],
+    static_libs: ["zlib_google_compression_utils_portable"],
     host_supported: true,
     test_suites: ["device-tests"],
 }
-- 
2.43.0

